Add `Lightning service providers` page to `How it works` section (#568)
BitcoinDesign/Guide: guide/designing-products/lightning-service-providers.md

---
layout: guide
title: Lightning Service Providers
description: Common user experience problems of the Lightning network that can be resolved by third-party services.
nav_order: 10
parent: Designing bitcoin products
permalink: /guide/designing-products/lightning-service-providers/
main_classes: -no-top-padding
---

<!--

Editor's notes

Illustration sources: https://www.figma.com/file/qr4P17z6WSPADm6oW0cKw2/?node-id=701%3A5911

-->

> {% include picture.html

Author: It's the actual image I had in mind to use, heavily based on this album from the movie drive: https://soundcloud.com/electricyouthmusic/college-and-electric-youth-a-real-hero 😆 

If people don't like it then I'm happy to change it

> {% include picture.html

Reviewer: I'm good with the style, but would agree with critique that the image only contains text and could be a better visual representation of the page content.

> {% include picture.html

Reviewer 2: I like the idea of a more creative, playful style, but the typeface looks a little Microsoft Windows 98, IMO. There are other script typefaces that might look nice.

- [Hurricane](https://fonts.google.com/specimen/Hurricane?category=Handwriting&preview.text=Lightning%20Service%20Providers&preview.text_type=custom)
- [Fontjek](https://www.dafont.com/fontjek.font?fpp=200&l[]=10&l[]=1&text=Lightning+Service+Providers)
- [Art Brush](https://www.dafont.com/artbrush.font?fpp=200&l[]=10&l[]=1&text=Lightning+Service+Providers)
- [Arkipelago](https://www.dafont.com/arkipelago.font?fpp=200&l[]=10&l[]=1&text=Lightning+Service+Providers)

At a bare minimum, just not putting most of the image text in all caps would help a lot. If they're going to be all caps, it kind of defeats the purpose of script typeface because a good script typeface is going to have these nice cursive connecting lines.

These are all nits.

> {% include picture.html

Author: > but the typeface looks a little Microsoft Windows 98

That's kind of the point 😆 I'll re-do the images if people don't like the aesthetic, these kind of things a pretty subjective though. The font used only has capitalization https://www.wfonts.com/font/mistral.

If you view my source file you can see the mood board I used for this style: https://www.figma.com/file/qr4P17z6WSPADm6oW0cKw2/?node-id=701%3A5911

   image = "/assets/images/guide/designing-products/lightning-service-providers/lightning-service-providers.jpg"
   retina = "/assets/images/guide/designing-products/lightning-service-providers/lightning-service-providers@2x.jpg"
   mobile = "/assets/images/guide/designing-products/lightning-service-providers/lightning-service-providers-mobile.jpg"
   mobileRetina = "/assets/images/guide/designing-products/lightning-service-providers/lightning-service-providers@2x.jpg"
   alt-text = "Lightning service providers header image."
   width = 1600
   height = 740
   layout = "full-width"
%}

> # Lightning Service Providers

Author: Yep I agree, good suggestion. Can leave it as is for now and have a bigger discussion around optimizing stuff like this in the future

> The lightning network is our best solution to bitcoins lack of scalability. However, using the lightning network self-custodially brings about many new usability issues, such as complex channel management and backups.

Author: Will add the links.

I just started using lowercase similar to how we are doing with bitcoin

> The lightning network is our best solution to bitcoins lack of scalability. However, using the lightning network self-custodially brings about many new usability issues, such as complex channel management and backups.

Reviewer 3: ```suggestion
The lightning network is our best solution to bitcoin's lack of scalability. However, using the lightning network self-custodially brings about many new usability issues, such as complex channel management and backups.
```

Lightning service providers (LSPs) aim to solve these issues. They act as non-custodial third parties that bitcoin lightning wallets can connect to and use services to automate various lightning operations.

> The goal of LSPs, [as put by Roy Sheinfeld](https://medium.com/breez-technology/envisioning-lsps-in-the-lightning-economy-832b45871992) of Breez, is to assist with onboarding the world to the Lightning economy.

Author: Will re-word this / make things more clear

## Lightning Services

> Lightning services are offered by LSPs to solve a particular pain point when using a non-custodial bitcoin lightning wallet.

Author: > I think the point is that when the wallet is custodial, you don't need an LSP

Yes this is correct. I can see how this could be confusing, will re-word it so it's more clear. LSPs are non-custodial yes, they are only used as 'hand-holders' for custodial wallets but never take custody. Some LSPs do however, say when doing a trusted swap. But they could (should) be using submarine swaps which are non-custodial. Attempted to stress that point in the swap sections of this page

   image = "/assets/images/guide/designing-products/lightning-service-providers/inbound-liquidity.jpg"
   retina = "/assets/images/guide/designing-products/lightning-service-providers/inbound-liquidity@2x.jpg"
   alt-text = "Drop of water representing liquidity."
   width = 140
   height = 140
   layout = "float-right"
%}

Lightning wallets can not receive payments without inbound liquidity in a channel. This is primarily the case with new wallets, but may also occur when channels are imbalanced on funded wallets.

Offering inbound liquidity with a channel open service addresses this situation. When a user receives a payment without enough inbound liquidity, a channel is automatically opened with some. The network, and sometimes a service, fee for opening the channel is typically deducted from the payment amount.

A channel open with inbound liquidity allows users to fund their wallets with a lightning payment, or combined with a [swap-in]({{ '/guide/designing-products/lightning-service-providers/#swap-in' | relative_url }}) or [dual funding]({{ '/guide/designing-products/lightning-service-providers/#dual-funding' | relative_url }}) service can allow funding via on-chain.

> [Blocktank](https://blocktank.synonym.to/) and [Dunder](https://github.com/hsjoberg/dunder-lsp) are two open source LSP clients that help applications offer inbound liquidity to their users.

Author: Yeah it is https://github.com/synonymdev/backpack-host

A [dual-funding](https://bitcoinops.org/en/topics/dual-funding/) channel open service solves this issue of trust by having users cooperatively open a channel with the LSP. Users get the benefits of having both inbound and outbound liquidity without having to trust the LSP.

Dual funding is only applicable when funding a wallet with on-chain bitcoin, the issue of trusting the LSP still exists if funding with a lightning payment.

</div>

### Spend-unconfirmed

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/designing-products/lightning-service-providers/spend-unconfirmed.jpg"
   retina = "/assets/images/guide/designing-products/lightning-service-providers/spend-unconfirmed@2x.jpg"
   alt-text = "Blocks representing unconfirmed spend."
   width = 140
   height = 140
   layout = "float-right"
%}

When a channel is opened, the bitcoin can not be spent until the on-chain channel open transaction is confirmed. If the mempool is congested, this wait could take a long time.

> A channel open service using spend-unconfirmed, sometimes called turbo channels, allows users to spend their bitcoin instantly without having to wait for the channel transaction to confirm.

Reviewer 4: ```suggestion
A channel open service using spend-unconfirmed, sometimes called turbo channels or zero-confirmation channels allows users to spend their bitcoin instantly without having to wait for the channel transaction to confirm.
```

> A spend-unconfirmed channel is trusted until the transaction is confirmed. Users should have the option to disable these type of channel opens if they are used. Some LSPs may batch their channel opens meaning a users channel will be trusted until this is done.

Reviewer 2: ```suggestion
A spend-unconfirmed channel is trusted until the transaction is confirmed. Users should have the option to disable these types of channel opens if they are used. Some LSPs may batch their channel opens meaning a users channel will be trusted until this is done.
```

[Breez](https://breez.technology/) is an bitcoin wallet that uses this service.

</div>

### Swap-in

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/designing-products/lightning-service-providers/swap-in.jpg"
   retina = "/assets/images/guide/designing-products/lightning-service-providers/swap-in@2x.jpg"
   alt-text = "Swap into lightning."
   width = 140
   height = 140
   layout = "float-right"
%}

Lightning wallet need local on-chain bitcoin to open a payment channel. Having to fund a lightning wallet on-chain, then use those bitcoin to open a channel adds a lot of friction to onboarding. This is particularly the case for new users who likely only have on-chain bitcoin.

A swap-in channel open service solves this by allowing users to fund a lightning wallet with an on-chain payment. The user sends their on-chain bitcoin to the LSP who opens a channel for the user.

> Swap-ins can be done in a trusted way, or trust-minimized way using [submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/). A [dual-funded]({{ '/guide/designing-products/lightning-service-providers/#dual-funding' | relative_url }}) channel open is a more trust-minimized way to resolve this issue.

Reviewer 2: ```suggestion
Swap-ins can be done in a trusted way or a trust-minimized way using [submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/). A [dual-funded]({{ '/guide/designing-products/lightning-service-providers/#dual-funding' | relative_url }}) channel open is a more trust-minimized way to resolve this issue.
```

The client [Loop](https://lightning.engineering/loop/) and the non-custodial lightning wallet [Phoenix](https://phoenix.acinq.co/) offer this service.

</div>

### Zero-reserve

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/designing-products/lightning-service-providers/zero-reserve.jpg"
   retina = "/assets/images/guide/designing-products/lightning-service-providers/zero-reserve@2x.jpg"
   alt-text = "Empty glass representing no channel reserve."
   width = 140
   height = 140
   layout = "float-right"
%}

> Payment channels require a channel reserve of 1% to be set aside to prevent theft. When a user opens a channel, they will not be able to spend this reserve. This can confuse many users who may think an incorrect payment amount was sent to them.

Author: BIP2 defines it as 1%, removed the 1% and just mentioned that require one as it does differ across implementations.

> A zero-reserve channel open service solves this issue by allowing channels opened by the LSP to have a zero-reserve on the users side. The LSP still has a reserve as it is connected to the wider network. This also prevents them from attempting to steal the users bitcoin.

Reviewer 2: ```suggestion
A zero-reserve channel open service solves this issue by allowing channels opened by the LSP to have a zero-reserve on the users side. The LSP still has a reserve as it is connected to the wider network. This also prevents them from attempting to steal the user's bitcoin.
```

   layout = "float-right-desktop"
%}

Lightning [payment requests]({{ '/guide/payments/request/payment-request-formats/' | relative_url }}) are not very human-readable and are mostly [single-use]({{ '/guide/payments/request/#single-use-payment-requests/' | relative_url }}). Sharing an invoice for every payment adds a lot of friction when sending and requesting payments.

A [Lightning address](https://lightningaddress.com/) service gives users a reusable, human-readable way to receive lightning payments, such as bosch@bitcoin.design.

Lightning addresses use a non-custodial, trusted third-party server to forward payment requests. This opens up the possibility that the LSP could censor user payments on the application layer by not relaying them between the parties.

</div>

## Payment forwarding

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/designing-products/lightning-service-providers/payment-forwarding.jpg"
   retina = "/assets/images/guide/designing-products/lightning-service-providers/payment-forwarding@2x.jpg"
   alt-text = "Lightning address LSP service."
   width = 400
   height = 300
   layout = "float-right-desktop"
%}

> Lightning payments require a route through the network to be calculated for sending, and a user must be online for receiving. Mobile nodes, especially ones running on lower end devices, may have difficulties calculating a route which can result in higher network fees. They are also regularly offline meaning a user will have to be online to receive a payment.

Reviewer 4: ```suggestion
Lightning payments require a route through the network to be calculated for sending, and a user must be online for receiving. Mobile nodes, especially ones running on lower-end devices, may have difficulties calculating a route, resulting in higher network fees. They are also regularly offline meaning a user will have to be online to receive a payment.
```

Reviewer: Another comma here `...offline, meaning...`

> A payment forwarding service solves these issues by acting as an non-custodial intermediary for payments. The route calculation can be conducted by the LSPs node on behalf of the user for sending. And for receiving if the user is offline the LSP can hold the payment and forward it once the user comes online.

Reviewer: ```suggestion
A payment forwarding service solves these issues by acting as an non-custodial intermediary for payments. The route calculation for sending bitcoin can be conducted by the LSPs node on behalf of the user. And for receiving, if the user is offline, the LSP can hold the payment and forward it once the user comes back online.
```

> A payment forwarding service solves these issues by acting as an non-custodial intermediary for payments. The route calculation for sending bitcoin can be conducted by the LSPs node on behalf of the user. And for receiving, if the user is offline, the LSP can hold the payment and forward it once the user comes back online.

Reviewer 2: ```suggestion
A payment forwarding service solves these issues by acting as a non-custodial intermediary for payments. The route calculation for sending bitcoin can be conducted by the LSPs node on behalf of the user. And for receiving, if the user is offline, the LSP can hold the payment and forward it once the user comes back online.
```

[Trampoline payments](https://bitcoinops.org/en/topics/trampoline-payments/) remove the need for a single third-party node to calculate a payment route for a user. However, these are currently not widely supported.

[Phoenix](https://phoenix.acinq.co/) is a non-custodial lighting wallet offering this service.

</div>

> ## Swap-out

Reviewer 4: ```suggestion
### Swap-out
```

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/designing-products/lightning-service-providers/swap-out.jpg"
   retina = "/assets/images/guide/designing-products/lightning-service-providers/swap-out@2x.jpg"
   alt-text = "Swapping out of the lightning network."
   width = 400
   height = 300
   layout = "float-right-desktop"
%}

Making a payment directly from lightning to on-chain is not possible. A user may want to move funds from their lightning channel to on-chain for more secure storage.

> A swap-out service allows users to send bitcoin from lightning to on-chain. Users send their lightning bitcoin to the LSP. They then make the on-chain payment on the users behalf. This can have added privacy benefits as the user is receiving one of the LSPs UTXOs and not the ones in their channel.

Reviewer 2: ```suggestion
A swap-out service allows users to send bitcoin from lightning to on-chain. Users send their lightning bitcoin to the LSP. They then make the on-chain payment on the user's behalf. This can have added privacy benefits as the user is receiving one of the LSPs UTXOs and not the ones in their channel.
```

> Swap-outs can be done in a non-custodial, trust-minimized way using [Submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/). They can also be done in a trusted way where the LSP has temporary custody of the funds, this is no recommended as the LSP could steal the users bitcoin or censor their payment.

Reviewer: ```suggestion
Swap-outs can be done in a non-custodial, trust-minimized way using [Submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/). They can also be done in a trusted way where the LSP has temporary custody of the funds. This is no recommended, as the LSP could steal the users bitcoin, or censor their payment.
```

> Swap-outs can be done in a non-custodial, trust-minimized way using [Submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/). They can also be done in a trusted way where the LSP has temporary custody of the funds. This is not recommended, as the LSP could steal the users bitcoin, or censor their payment.

Reviewer 2: ```suggestion
Swap-outs can be done in a non-custodial, trust-minimized way using [Submarine swaps](https://blog.muun.com/a-closer-look-at-submarine-swaps-in-the-lightning-network/). They can also be done in a trusted way where the LSP has temporary custody of the funds. This is not recommended, as the LSP could steal the user's bitcoin, or censor their payment.
```

> [Lightning labs](https://lightning.engineering/loop/) is an example of a LSP using a submarine swap-out service called loop.

Reviewer 2: ```suggestion
[Lightning Labs](https://lightning.engineering/loop/) is an example of an LSP using a submarine swap-out service called loop.
```

</div>

## Watchtowers

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/designing-products/lightning-service-providers/watchtowers.jpg"
   retina = "/assets/images/guide/designing-products/lightning-service-providers/watchtowers@2x.jpg"
   alt-text = "Lightning address LSP service."
   width = 400
   height = 300
   layout = "float-right-desktop"
%}

Lightning nodes need to stay online in order to monitor for potential cheating attempts from their counterparties. Mobile nodes in particular are regularly offline, meaning their channel counterparties could potentially cheat them and steal their bitcoin.

> A [watchtower](https://bitcoinops.org/en/topics/watchtowers/) service prevents this by monitoring the users node for cheating attempts. If one is detected, a watchtower broadcasts a justice transaction which gives the cheating parties bitcoin to the honest user. Watchtowers sometimes take a small fee for offering this service.

Reviewer 2: ```suggestion
A [watchtower](https://bitcoinops.org/en/topics/watchtowers/) service prevents this by monitoring the user's node for cheating attempts. If one is detected, a watchtower broadcasts a justice transaction which gives the cheating parties bitcoin to the honest user. Watchtowers sometimes take a small fee for offering this service.
```

> When using a watchtower and a LSP for other services, it's important to have users connect to one that isn't the LSP. This is due to the LSP likely being the only counterparty so having a watchtower with them would not safeguard the users bitcoin.

Reviewer: ```suggestion
When using a watchtower and a LSP for other services, it's important to have users connect to one that isn't the LSP. This is due to the LSP likely being the only counterparty, so having a watchtower with them would not safeguard the users bitcoin.
```

> When using a watchtower and a LSP for other services, it's important to have users connect to one that isn't the LSP. This is due to the LSP likely being the only counterparty, so having a watchtower with them would not safeguard the users bitcoin.

Reviewer 2: ```suggestion
When using a watchtower and a LSP for other services, it's important to have users connect to one that isn't the LSP. This is due to the LSP likely being the only counterparty, so having a watchtower with them would not safeguard the user's bitcoin.
```
